features/flows/rest/impl/src/main/java/org/opennms/netmgt/flows/rest/internal/classification/ClassificationRestServiceImpl.java

```diff
@@ -208,9 +201,37 @@ public Response getGroups(UriInfo uriInfo) {
     }
 
     @Override
-    public Response getGroup(int groupId) {
+    public Response getGroup(int groupId, String format, String requestedFilename, String acceptHeader) {
+
+        boolean isCsvRequested = StringUtils.contains(acceptHeader, "text/comma-separated-values")
+                || StringUtils.equalsIgnoreCase( "csv", format);
+
+        if(isCsvRequested
+                && requestedFilename != null
+                && !new FilenameHelper().isValidFileName(requestedFilename)) {
+          return Response.status(Response.Status.BAD_REQUEST).entity("parameter filename should follow this regex pattern: "
+                  + FilenameHelper.REGEX_ALLOWED_CHAR).build();
+
+        } else if(isCsvRequested) {
+            final String csvContent = classificationService.exportRules(groupId);
+            final String filename = new FilenameHelper().createFilenameForGroupExport(groupId, requestedFilename);
+            return Response.ok()
+                    .header("Content-Disposition", "attachment; filename=\""+filename+"\"")
+                    .header("Content-Type", "text/comma-separated-values")
+                    .entity(csvContent)
+                    .build();
+
+        } else {
+            return getGroupAsJson(groupId);
```

Comment:
Can we either create a `getGroupAsCsv` method or not have a `getGroupAsJson` either way is fine.